torch/csrc/autograd/engine.cpp

```diff
@@ -410,6 +441,22 @@ auto Engine::evaluate_function(FunctionTask& task) -> void {
       is_ready = true;
     }
 
+    // Switches to and sync with child's stream (if needed)
+    #ifdef USE_CUDA
+      // Sets child's device and stream
+      const int child_device = output.is_cuda() ? output.get_device() : - 1;
+      auto child_stream = next.function->input_metadata(next.input_nr).stream();
+      AutoGPUStream child_auto_gpu_stream{child_device, child_stream};
```

Comment:
This always does a bunch of stream operations, even if they're completely unnecessary (e.g. because the stream is the same as the one used in the function call. It would be better to reuse `auto_gpu_stream` from above and do sth like `auto_gpu_stream.setStream(child_stream)`

Reply:
Reusing makes sense, but it has to be .set(child_device, child_stream).